src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/InitializeNrGeneration.java

```diff
@@ -0,0 +1,47 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+
+public class InitializeNrGeneration {
+
+    public static int countNumOfOp( final LogicalPlan plan ) {
+        final LogicalOperator lop = plan.getRootOperator();
+
+        if ( lop instanceof LogicalOpMultiwayJoin) {
+            final int count = plan.numberOfSubPlans();
+
+            int nrGeneration = calcFactorial(count);
+            for ( int i = 0; i < count; i++ ) {
+                nrGeneration = nrGeneration * countNumOfOp( plan.getSubPlan( i ) );
+            }
+            return nrGeneration > 0 ? nrGeneration: Integer.MAX_VALUE;
```

Comment:
I am afraid that you need to do this check each time you multiply. Otherwise, you cannot catch cases in which the integer became negative and then became positive again.

Alternatively, and probably better, as soon as the value becomes negative **inside** the for-loop, return Integer.MAX_VALUE.

Reply:
Thanks, fixed now.